### PR TITLE
operator == for string view

### PR DESCRIPTION
**Pull request by RadekVana**

### CODE REVIEW

File: src/lang/string_view_utf8.hpp [hunk 1]
```diff
@@ -221,4 +221,27 @@ class string_view_utf8 {
         s.type = EType::NULLSTR;
         return s;
     }
+
+    /// string view has same resource
+    constexpr bool operator==(const string_view_utf8 &other) const {
+        if (type != other.type)
+            return false; // type missmatch
```
Comment (DRracer) [inline src/lang/string_view_utf8.hpp:228]: mismatch (one 's')

File: src/lang/string_view_utf8.hpp [hunk 2]
```diff
@@ -221,4 +221,27 @@ class string_view_utf8 {
         s.type = EType::NULLSTR;
         return s;
     }
+
+    /// string view has same resource
```
Comment (DRracer) [inline src/lang/string_view_utf8.hpp:225]: has THE same resource

File: src/lang/string_view_utf8.hpp [hunk 3]
```diff
@@ -221,4 +221,27 @@ class string_view_utf8 {
...
+        if (type != other.type)
+            return false; // type missmatch
+
+        switch (type) {
+        case EType::RAM:
+        case EType::CPUFLASH:
+            return attrs.cpuflash.utf8raw == other.attrs.cpuflash.utf8raw;
+        case EType::FILE:
+            return (attrs.file.f == other.attrs.file.f) && (attrs.file.startOfs == other.attrs.file.startOfs);
+        case EType::SPIFLASH:
```
Comment (DRracer) [inline src/lang/string_view_utf8.hpp:236]: SPIFLASH and USBFLASH types are equal by default as well? That's not true

**Review by DRracer — CHANGES_REQUESTED** [review]
please fix:
- some typos in comments
- comparison of usb and spiflash string views (even though there are no string views like this yet)

File: src/lang/string_view_utf8.hpp (hunk 3 above)
Comment (RadekVana) [reply src/lang/string_view_utf8.hpp:237]: added assert

File: src/lang/string_view_utf8.hpp (hunk 2 above)
Comment (RadekVana) [reply src/lang/string_view_utf8.hpp]: fixed

File: src/lang/string_view_utf8.hpp (hunk 1 above)
Comment (RadekVana) [reply src/lang/string_view_utf8.hpp]: fixed

**Review by DRracer — APPROVED** [review]

**Pull request merged**